validphys2/src/validphys/coredata.py

```diff
@@ -68,7 +68,7 @@ class FKTableData:
     ndata: int
     xgrid: np.ndarray
     sigma: pd.DataFrame
-    convolution_types: tuple[str] = None
+    convolution_types: list | None = None
```

Comment:
Tuple would be best so it can be hashed if needed (not 100% important since in principle one given fktable will have only one entry for this I guess)

Reply:
I agree!

On a side note, having `list[pineappl.convolutions.Conv]` or `tuple[pineappl.convolutions.Conv]` might not work for hyperopt (?) because the `Conv` object is not pickleable.

Reply:
I don't think the fktables go down the pickle with hyperopt, do they?

Because if they do we just found the memory leak :__

Reply:
They shouldn't! But somehow the hyperopt is failing because it cannot pickle `Conv`:
https://github.com/NNPDF/nnpdf/actions/runs/11896970990/job/33150122459?pr=2217#step:6:2168

Not really sure what is happening here.

Reply:
It should certainly not. Good we found that ^^

Reply:
I have looked a bit at this. We definitely should prune what goes into the pickle there _however_ I think the parser should take the pineappl object and convert them to strings. We don't want to have pineappl objects during the training in the fit since we want to be able to act exclusively with keras + numpy objects, sobe  the `FKTableData` should definitely keep the `convolution_types: tuple[str] = None`

Reply:
Thanks for looking into this!

> We don't want to have pineappl objects during the training in the fit since we want to be able to act exclusively with keras + numpy objects. The FKTableData should definitely keep the convolution_types: tuple[str] = None

If the state of the `FKTableData` object is still passed onto the fit, then I agree that `convolution_types` should be `tuple[str]`. But to be honest, I didn't expect this to be the case 👀

So, in the meantime, yes I will do as so.

Reply:
Well, either at the parser or later we need to parser `convolution_types` because we need to know which tensorflow function to use. So, since it needs to be parsed anyway, it should be done as soon as possible so that we are always using the parsed values instead of having parts of the code that use the pineappl value and part of the code that use the strings.